consent-ws/src/main/java/org/broadinstitute/consent/http/resources/ConsentResource.java

```diff
@@ -78,6 +79,9 @@ public Response createConsent(@Context UriInfo info, Consent rec, @Auth User use
             if (rec.getDataUse() == null) {
                 throw new IllegalArgumentException("Data Use Object is required.");
             }
+            if (rec.getDataUseLetter() != null) {
```

Comment:
`checkValidDUL()` checks `if (rec.getDataUseLetter() != null)` so no need for duplicating it here? Similarly for line#111.

Alternatively, `if (rec.getDataUseLetter() != null)` could be removed from `checkValidDUL()` if that makes more sense.

Reply:
Good point - I think this double-check a result of some refactoring. Will definitely clean this up some more.